adapter/src/hermes/adapter/stdio/mapper/balanced_mapper.cc

```diff
@@ -0,0 +1,33 @@
+#include "balanced_mapper.h"
+
+/**
+ * Namespace declaration for cleaner code.
+ */
+using hermes::adapter::stdio::BalancedMapper;
+using hermes::adapter::stdio::FileStruct;
+using hermes::adapter::stdio::HermesStruct;
+
+MapperReturnType BalancedMapper::map(const FileStruct& file_op) {
+  LOG(INFO) << "Mapping File with offset:" << file_op.offset_
+            << " and size:" << file_op.size_ << "." << std::endl;
+
+  auto mapper_return = MapperReturnType();
+  size_t size_mapped = 0;
+  while (file_op.size_ > size_mapped) {
+    FileStruct file;
```

Comment:
Do you need to copy `file_op.file_id_` into `file.file_id_` here? Every mapping returned from this function contains the default `FileID` of `{0,0}`. Is that a bug or intentional?

Reply:
It doesn't have any impact on the logic. But your right, I will add it either way.

Reply:
If the logic doesn't depend on the `FileID`, then how do you detect if two processes open the same file?

Reply:
That is outside of this. It is done during open. And a respective bucket is opened/created. At this stage, the mapper just maps offset and size to blobs.

Reply:
Ok.